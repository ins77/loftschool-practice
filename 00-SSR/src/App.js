import React from "react";
import "./App.css";
import { Switch, Redirect, Route, Link } from "react-router-dom";

function Home() {
  return <h1>Home</h1>
}

function About() {
  return <h1>About</h1>
}

function RedirectPage() {
  return <Redirect to="/"/> 
}

function App() {
  return (
    <>
    {/* <Redirect to="https://google.com"/> */}
    <Link to="/home">Home</Link>
    <Link to="/about">About</Link>
    <Link to="/redirect">Redirect</Link>
    <Switch>
      <Route path="/home" component={Home}/>
      <Route path="/about" component={About}/>
      <Route path="/redirect" component={RedirectPage}/>
    </Switch>
    </>
  );
}

export default App;
