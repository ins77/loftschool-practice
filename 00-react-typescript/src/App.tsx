import React from "react"
import Column from "./Column"
import CustomDragLayer from "./CustomDragLayer"
import { useItemsContext } from "./ItemsContext/ItemsContext"
import { AppContainer } from "./styles"
import { AddNewItem } from "./AddNewItem"

const App = () => {
  const { state, dispatch } = useItemsContext()

  return (
    <AppContainer>
      <CustomDragLayer />
      {state.lanes.map((column, i: number) => {
        return (
          <Column key={column.id} id={column.id} text={column.text} index={i} />
        )
      })}
      <AddNewItem
        toggleButtonText="+ Add another list"
        dispatchButtonText="Add list"
        onAdd={text => dispatch({ type: "ADD_LANE", payload: text })}
      />
    </AppContainer>
  )
}

export default App
