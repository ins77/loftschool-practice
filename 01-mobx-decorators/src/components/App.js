import React from "react";
import MainContent from "./MainContent";
import Footer from "./Footer";

class App extends React.Component {

    render() {
        return (
            <React.Fragment>
                <MainContent/>
                <Footer/>
            </React.Fragment>
        );
    }
}

export default App;