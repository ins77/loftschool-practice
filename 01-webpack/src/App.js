import React from "react";
import twitter from './assets/twitter.svg';
import facebook from './assets/facebook.svg';
import wikipedia from './assets/wikipedia.svg';

const App = ({ title }) => (
  <div>
    <img
      width={twitter.width}
      height={twitter.height}
      src={twitter.url}
    />
    <img
      width={facebook.width}
      height={facebook.height}
      src={facebook.url}
    />
    <img
      width={wikipedia.width}
      height={wikipedia.height}
      src={wikipedia.url}
    />
    {title}
  </div>
);

export default App;
