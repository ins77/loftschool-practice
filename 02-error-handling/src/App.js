import React from 'react'

const styles = {
  fontFamily: "sans-serif",
  textAlign: "center"
};

class ErrorBoundary extends React.Component {
  state = { error: null, errorInfo: null };

  componentDidCatch(error, errorInfo) {
    this.setState({
      error: error,
      errorInfo: errorInfo
    });
  }

  render() {
    if (this.state.errorInfo) {
      return (
        <div>
          <h2>Something went wrong.</h2>
          <details style={{ whiteSpace: "pre-wrap" }}>
            {this.state.error && this.state.error.toString()}
            <br />
            {this.state.errorInfo.componentStack}
          </details>
        </div>
      );
    }

    return this.props.children;
  }
}

class Buggy extends React.Component {
 
  state = { greeting: "Welcome"};
  componentDidMount() {
    throw new Error("An error has occured in Buggy component!");
  }
  render() {
    return <h2>{this.state.greeting}</h2>;
  }
}

class NotBuggy extends React.Component {
 
  state = { greeting: "Welcome"};
  componentDidMount() {
   
  }
  render() {
    return <h2>{this.state.greeting}</h2>;
  }
}


const App = () => (
  <div style={styles}>

    <h2>Error Boundaries Example</h2>
    <ErrorBoundary>
      <Buggy />
    </ErrorBoundary>
    
    <ErrorBoundary>
      <NotBuggy />
    </ErrorBoundary>
    
  </div>
);

export default App;