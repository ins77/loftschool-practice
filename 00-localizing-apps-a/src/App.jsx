import React, { Component } from "react";
import bulle from "./bulle.png";
import Greeting from "./Greeting";
import "./App.css";

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={bulle} className="App-logo" alt="logo" />
          <Greeting />
        </header>
      </div>
    );
  }
}

export default App;
