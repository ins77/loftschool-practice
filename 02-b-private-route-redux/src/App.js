import React from "react";
import "./App.css";
import PrivateRoute from "./PrivateRoute";
import { Switch, Link, Route, useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";

const Home = () => {
  return <h1>Home</h1>;
};

const Private = () => {
  const dispatch = useDispatch();

  return (
    <>
      <h1>Private</h1>
      <button onClick={() => dispatch({ type: "LOGOUT" })}>Logout</button>
    </>
  );
};

const Login = () => {
  const dispatch = useDispatch();
  // const history = useHistory();

  return (
    <>
      <h1>Authenticate here</h1>
      <button
        onClick={() => {
          dispatch({ type: "LOGIN" });
          // history.push("/private");
        }}
      >
        Login
      </button>
    </>
  );
};

function App() {
  const authed = useSelector(state => state.authed);

  return (
    <>
      <Link to="/private">Private</Link>
      <Link to="/">Home</Link>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/login" component={Login} />
        <PrivateRoute exact path="/private" component={Private} />
      </Switch>
    </>
  );
}

export default App;
