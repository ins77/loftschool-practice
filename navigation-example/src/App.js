import React from "react";
import Profile from "./Profile";
import "./App.css";

const PAGES = {
  home: <>HOME</>,
  about: <>ABOUT</>,
  profile: <Profile />,
};

class App extends React.Component {
  state = { currentPage: this.props.initialPage };

  navigateTo = (page) => {
    this.setState({ currentPage: page });
  };

  render() {
    return (
      <>
        <nav>
          {Object.entries(this.props.pages).map((page) => {
            return (
              <button
                key={page[0]}
                onClick={() => {
                  this.navigateTo(page[0]);
                }}
              >
                {page[0]}
              </button>
            );
          })}
        </nav>
        <div data-testid="container">{this.props.pages[this.state.currentPage]}</div>
      </>
    );
  }
}

App.defaultProps = {
  pages: PAGES,
  initialPage: "home",
};

export default App;
