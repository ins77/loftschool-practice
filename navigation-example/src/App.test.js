import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import App from './App';

const PAGES = {
  foo: <>FOO</>,
  bar: <>BAR</>
}

describe("App", () => {
  it("renders correctly", () => {
    const {container } = render(<App pages={PAGES} initialPage="foo"/>)
    expect(container).toMatchSnapshot()
  })

  describe("when clicked on navigation buttons", () => {
    it("opens the corresponding page", () => {
      const {debug, getByText, getByTestId } = render(<App pages={PAGES} initialPage="foo"/>)
      const barButton = getByText("bar");
      // debug()
      fireEvent.click(barButton)
      // debug()
      expect(getByTestId("container")).toHaveTextContent("BAR")
    })
  })
})