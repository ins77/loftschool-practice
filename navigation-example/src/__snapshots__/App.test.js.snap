// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <nav>
    <button>
      foo
    </button>
    <button>
      bar
    </button>
  </nav>
  <div
    data-testid="container"
  >
    FOO
  </div>
</div>
`;
